modules/redhen_connection/redhen_connection.module

```diff
@@ -79,52 +78,34 @@ function redhen_connection_type_options_list() {
 
 /**
  * Implements hook_entity_access().
- *
- * @param \Drupal\Core\Entity\EntityInterface $entity
- * @param String $operation
- * @param \Drupal\Core\Session\AccountInterface $account
- *
- * @return mixed
  */
 function redhen_connection_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
-  if (in_array($entity->getEntityTypeId(), ['redhen_contact', 'redhen_org'])){
-    // Only check/grant access to active contacts and orgs.
-    if ($entity->isActive()) {
-      return ConnectionAccessResult::allowedIfHasConnectionPermission($entity, $account, $operation);
-    }
-  }
 
-  if ($entity->getEntityTypeId() == 'redhen_connection' && $entity->isActive()) {
-    // Allow access to a connection via indirectly connected redhen_contact
-    // entity if the contact associated with the current user is connected to
-    // the indirectly connected contact.
-
-    // Get redhen_contact endpoint fields from the connection type.
-    $connection_type = ConnectionType::load($entity->bundle());
-    $contact_endpoint_fields = $connection_type->getEndpointFields('redhen_contact');
-    // Loop over contact fields and get the indirectly connected contact entity.
-    foreach ($contact_endpoint_fields as $contact_endpoint_field) {
-      $contact = $entity->get($contact_endpoint_field)->entity;
-      // If we get a contact, check if the contact associated with the current
-      // user is connected to it, is active and has access.
-      if ($contact && $contact->isActive()) {
-        // @todo - add connection type to this call so we don't allow access if
-        // any connection type allows the passed operation - will require
-        // updating the function to take the additional parameter
-        return ConnectionAccessResult::allowedIfHasConnectionPermission($contact, $account, $operation);
+  $connection_plugin_manager = \Drupal::service('plugin.manager.connection_permission');
+  $plugin_definitions = $connection_plugin_manager->getDefinitions();
+
+  foreach ($plugin_definitions as $plugin_id => $plugin_definition) {
+    $plugin_instance = $connection_plugin_manager->createInstance($plugin_id);
+    if ($entity->getEntityTypeId() === $plugin_instance->get('subject_entity_type')) {
+      if (!empty($plugin_instance->get('subject_entity_bundle'))) {
+        if ($plugin_instance->get('subject_entity_bundle') != $entity->bundle()) {
+          continue;
+        }
+      }
+      $contact = Contact::loadByUser($account);
+      // @todo do we need to return neutral if no contact exists?
```

Comment:
If you have no return statement, you are returning "null" which is maybe fine, but the hook_entity_access documentation seems to say you should return a `AccessResult::neutral();` response: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/function/hook_entity_access/8.8.x.

Reply:
It also seems to indicate that you might break routes if they are using the same permissions, so if you try that out and suddenly can't access entity pages, we may need to look closer.